internal/testutils/jimmtest/fixture.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2024 Canonical.
+package jimmtest
+
+import (
+	"context"
+	"database/sql"
+	"time"
+
+	petname "github.com/dustinkirkland/golang-petname"
+	qt "github.com/frankban/quicktest"
+	"github.com/google/uuid"
+	"github.com/juju/juju/core/crossmodel"
+	"github.com/juju/juju/state"
+
+	"github.com/canonical/jimm/v3/internal/db"
+	"github.com/canonical/jimm/v3/internal/dbmodel"
+)
+
+// createTestControllerEnvironment is a utility function creating the necessary components of adding a:
+//   - user
+//   - user group
+//   - controller
+//   - model
+//   - application offer
+//   - cloud
+//   - cloud credential
+//   - role
+//
+// Into the test database, returning the dbmodels to be utilised for values within tests.
+//
+// TODO(ale8k): Make this an implicit thing on the JIMM suite per test & refactor the current state.
+// and make the suite argument an interface of the required calls we use here.
+func CreateTestControllerEnvironment(ctx context.Context, c *qt.C, db db.Database) (
+	dbmodel.Identity,
```

Comment:
this needs to return a struct type. there are way to many returns

Reply:
Agreed, but that's a refactoring for a separate PR.

Reply:
ALso this was temporary, wasn't supposed to be a catchall function when I did it but we ended up blasting it everywhere aha. We need a better way of doing this really